core/trino-main/src/main/java/io/trino/operator/OutputValidatingSourceOperator.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.operator;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.util.concurrent.ListenableFuture;
+import io.trino.metadata.Split;
+import io.trino.spi.Page;
+import io.trino.spi.type.Type;
+import io.trino.sql.planner.plan.PlanNodeId;
+
+import java.util.List;
+import java.util.function.Supplier;
+
+import static io.trino.operator.PageValidations.validateOutputPageTypes;
+import static java.util.Objects.requireNonNull;
+
+public class OutputValidatingSourceOperator
+        implements SourceOperator
```

Comment:
It's a forwarding class.
Add a test verifying that it implements all interface methods.